sections/performance/block-loop.md

```diff
@@ -0,0 +1,73 @@
+# Don't block the event loop
+
+<br/><br/>
+
+Node handles the Event Loop mostly on a single thread rotating through multiple queues. Operations with high complexity, large json parsing, unsafe regex querieies, and large IO operations are some of the operations that can cause the Event Loop to stall. Avoid this off-loading CPU intensive tasks to a dedicated service (e.g. job server), or breaking long tasks into small steps then using the Worker Pool are some examples of how to avoid blocking the Event Loop.
```

Comment:
Let's now take it one step up and make our advice even more accurate, I'll be glad to explain why is this true:

"When handling CPU-intensive tasks, different techniques should be applied based on of two different scenarios: if these CPU-intensive tasks are needed rarely (e.g. trigerred by a few requests) - then they can be offloaded to a dedicate process, thread or get handled in chunks (See example below). However, the more challenging case is where most requests need to trigger CPU-intensive code -  in that case, this task must be offloaded to a different workload, one that doesn't dispatch content to the end-users, like a job-server Serverless, and similar

Reply:
Sounds good to me and makes sense. Should I add this as a separate paragraph?

Reply:
Yes, or maybe to the main one?